pykafka/simpleconsumer.py

```diff
@@ -646,20 +655,18 @@ def _handle_success(parts):
                         min_bytes=self._fetch_min_bytes
                     )
                 except (IOError, SocketDisconnectedError):
+                    if self._running:
+                        unlock_partitions(iterkeys(partition_reqs))
```

Comment:
As soon as the lock is acquired, the partition is added as a key to `partition_reqs` with `None` as its value. See [these lines](https://github.com/Parsely/pykafka/blob/3026e9c1858ed1844cf66530c05bc8353087a6b6/pykafka/simpleconsumer.py#L635-L636) and the fact that they are right next to each other. I've been working under the assumption that since we don't further mutate that dict until the next loop iteration, we're guaranteed that we're unlocking exactly the set of locked partitions. Maybe the style of this code could use some clarifying, but I don't see a bug in it at least.